src/components/MoneyTemporaryForRefactorRequestConfirmationList.js

```diff
@@ -284,9 +284,9 @@ function MoneyTemporaryForRefactorRequestConfirmationList({
     const shouldShowBillable = !lodashGet(policy, 'disabledFields.defaultBillable', true);
 
     const hasRoute = TransactionUtils.hasRoute(transaction);
-    const isDistanceRequestWithoutRoute = isDistanceRequest && !hasRoute;
-    const formattedAmount = isDistanceRequestWithoutRoute
-        ? translate('common.tbd')
+    const isDistanceRequestWithPendingRoute = isDistanceRequest && (!hasRoute || !rate);
```

Comment:
Why isn't this using `isFetchingWaypointsFromServer()`?

Reply:
The `isFetchingWaypointsFromServer` depends on this variable, not vice versa:

https://github.com/Expensify/App/blob/e63553c39a3284902c46125a9e2fcae0e421ec22/src/components/MoneyTemporaryForRefactorRequestConfirmationList.js#L477-L479

Reply:
I feel like that's kind of an anti pattern. I think we should only call actions based directly on user action, not based on data changing. Do you agree Tim?

Reply:
But we already do call these actions in multiple places when landing on different steps of the IOU creation. Since in all these places we update the **draft** transaction, it looks more like a way to transfer the state between the IOU creation steps, and not an actual data modification.

Reply:
Ah, I think you're both right. @paultsimura you're correct that this is more akin to storing the form state between steps. @neil-marcellini I think you might be onto something about this looking like an anti-pattern.

What do you think about this?
- Instead of calling `IOU.setMoneyRequestPendingFields(transaction.transactionID, {waypoints: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD});` on this page, set that data on the previous page when the user submits the form. I think it could be part of the data that gets set here: https://github.com/Expensify/App/blob/e83c24a01a31607e5ccf406de8dd09a22c9d7c7e/src/pages/iou/request/step/IOURequestStepDistance.js#L159

Reply:
The issue here is the following scenario:

The user completes the first steps up until the confirmation step while being offline. Then, on the confirmation step, they come back online. Here, the map should get fetched, and we no longer should have the waypoints pending (even though when we landed on this page, the map was still pending).

Reply:
Oh, that darn offline flow gets confusing! Thanks for point that out. It's probably OK then, but I would ask that you also put that context into a code comment where it happens.

Reply:
Got you, thanks for taking your time on this. Updated ✔️

Reply:
Wow yeah that's a very interesting offline case. Thanks for explaining it and adding the comment.

Reply:
> Then, on the confirmation step, they come back online. Here, the map should get fetched, and we no longer should have the waypoints pending

Thank you for explaining. I think that flow makes sense, but how does that relate to this condition? If we are offline and set the waypoints pending when they are updated (as Tim suggested), then when the user is navigated to this page the pending fields will show correctly. Once they go back online I would think that the GetRoute response should clear the pending fields for the waypoints (and or the route), and then all data on the confirmation page would no longer show as pending. Does that make sense?

I'm going to leave this as a NAB. We could fix it in a follow up if needed.

Reply:
Probably yes, but I'm not sure where to attach the workspace rate being still loaded in this case. That being said, I would really appreciate it if we could merge this PR as-is if it's an NAB. It's my longest one so far 🙂